colossalai/engine/paramhooks/_base_paramhook.py

```diff
@@ -0,0 +1,37 @@
+from typing import Callable, List
+import torch
+
+class BaseParamHook(object):
+    def __init__(self, param_list: List[torch.nn.Parameter], backward_hook_call: Callable) -> None:
+        r"""
+        register backward hook on every parameters of module
+        """
+        self.param_list = param_list
+        self._register_backward_hooks(backward_hook_call)
+
+    def _register_backward_hooks(self, hook_call : Callable) -> None:
+        r"""
+        The hook_call will be called every time a gradient with respect to the a param in self.param_list 
+        is computed. 
+        The hook should have the following signature:
+        ```
+        hook(grad) -> Tensor or None
```

Comment:
I think the hook may be parameter relevant. Can this hook receive a parameter as argument?

Reply:
Sure. You define a hook function in this way and add parameters as many as you need in a wrapper class.
```
class HooKWrapper:
        def __init__(self) -> None:
            self.hook_triggered_times = 0

        def wrapper_func(self):
            def hook(grad) -> torch.Tensor or None:
                self.hook_triggered_times += 1
                return grad
            return hook
```